microprofile/security/src/main/java/io/helidon/microprofile/security/SecurityCdiExtension.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright (c) 2019 Oracle and/or its affiliates. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.helidon.microprofile.security;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.event.Observes;
+import javax.enterprise.inject.spi.BeforeBeanDiscovery;
+import javax.enterprise.inject.spi.Extension;
+
+/**
+ * Extension to register bean {@link SecurityProducer}.
+ */
+public class SecurityCdiExtension implements Extension {
+    void registerBean(@Observes BeforeBeanDiscovery abd) {
+        abd.addAnnotatedType(SecurityProducer.class, "helidon-security-producer")
+                .add(ApplicationScoped.Literal.INSTANCE);
```

Comment:
Just a comment, no action required: you can do this here, or you can just put `@ApplicationScoped` on `SecurityProducer`.

Reply:
This is intentional - I am not annotating any class with a bean defining annotation. This is to help with uses that create a fat jar. In such a case, this class would be registered twice (once through extension, once due to annotation scanning).